### ISSUE TITLE
Static analysis: dealii-git/source/fe/fe_q_hierarchical.cc

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#3406]
```
dealii-git/source/fe/fe_q_hierarchical.cc	505	warn	V688 The 'degree' local variable possesses the same name as one of the class members, which can result in a confusion.
```

We should address these warnings and errors from the static analysis tool PVS. In response to #3342.

**Comment by class4kayaker** [comment]
Variable is a convenience duplication of the class member. Possible solutions are to make an adjustment to silence this, but the confusion referenced in the warning is the intended implication.

**Comment by bangerth** [comment]
The local variable seems pretty pointless, and duplication is always bad. I would just get rid of it and replace it by `this->degree` instead.

**Comment by bangerth** [comment]
Fixed by #3444.
